### PR TITLE
Improve AI question generation failure display

### PR DESCRIPTION
**Pull request by jlevine18**
See https://github.com/PrairieLearn/PrairieLearn/pull/10422#discussion_r1711784306. Merge #10422 first.

### CODE REVIEW

**Comment by nwalters512** [comment]
#10422 landed, this is ready to be brought up to date with `master`.

**Comment by jlevine18** [comment]
> #10422 landed, this is ready to be brought up to date with `master`.

Done with merge commit ee96ca22cc52e165b27b7e375f6b4b6ba9c7d3ba.

File: apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.html.ts [hunk 1]
```diff
@@ -135,3 +135,14 @@ ${generatedPython}
     </div>
   `.toString();
 };
+
+export const GenerationFailure = (urlPrefix: string, jobSequenceId: string): string => {
```
Comment (nwalters512) [inline apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.html.ts:139]:
```suggestion
export function GenerationFailure(urlPrefix: string, jobSequenceId: string): string {
```

Let's also use an object with named properties, just like we do everywhere else.

File: apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts [hunk 2]
```diff
@@ -161,13 +165,7 @@ router.post(
...
         );
       } else {
-        if (req.get('HX-Request')) {
-          res
-            .set('HX-Redirect', res.locals.urlPrefix + '/jobSequence/' + result.jobSequenceId)
-            .send();
-        } else {
-          res.redirect(res.locals.urlPrefix + '/jobSequence/' + result.jobSequenceId);
-        }
+        res.send(GenerationFailure(res.locals.urlPrefix, result.jobSequenceId));
```
Comment (nwalters512) [inline apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts:168]: This only updates the regeneration flow; how about for initial generations?

File: apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.html.ts (hunk 1 above)
Comment (jlevine18) [reply apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.html.ts]: done in 4c35422

File: apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts (hunk 2 above)
Comment (jlevine18) [reply apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts:162]: Done in 4c35422.

File: apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.html.ts (hunk 1 above)
Comment (jlevine18) [reply apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.html.ts]: done in e3bef33

**Review by nwalters512 — APPROVED** [review]

**Pull request merged**